src/include/units/quantity.h

```diff
@@ -237,16 +239,34 @@ class quantity {
 
   // Hidden Friends
   // Below friend functions are to be found via argument-dependent lookup only
-  [[nodiscard]] friend constexpr quantity operator+(const quantity& lhs, const quantity& rhs)
-    requires invoke_result_convertible_to_<rep, std::plus<>, rep, rep>
+  template<typename Value>
+  [[nodiscard]] friend constexpr Quantity auto operator+(const quantity& lhs, const Value& rhs)
+    requires (!Quantity<Value>) && is_same_v<unit, units::one> &&
```

Comment:
`is_same_v<dimension, dim_one>`? Maybe a custom helper for `is_same_v<dimension, dim_one> && is_same_v<unit, ::units::one>` should be introduced?

Reply:
`is_same_v<dimension, dim_one>` is redundant. The `UnitOf<D>` constraint of `quantity` takes care of that. Alternatively, these could be non-`friend`s taking a `dimensionless<units::one, Rep>`, like here:
https://github.com/mpusz/units/blob/4051a745543f3243e7ae5af8a6d789a12f81b7cd/src/include/units/quantity.h#L214

Reply:
`units::one` can also be used for other dimensions (not only `dimensionless`):

https://github.com/mpusz/units/blob/a8fccf0cb5208385fd173e68db625dbbdf13b8a3/src/include/units/physical/natural/bits/dimensions.h#L43

Reply:
Although, I start to think that the lack of dimension check can actually help here? `dim_speed` in the natural units is really like dimensionless quantity. So in such a case we should remove checking for the dimension in other places where we check for `one`. For example:

https://github.com/mpusz/units/blob/a8fccf0cb5208385fd173e68db625dbbdf13b8a3/src/include/units/quantity.h#L136

Reply:
Just to be consistent

Reply:
So it wasn't redundant, but naturally allowed other use cases. I'll open an issue for that.